apps/prairielearn/src/lib/authn.js

```diff
@@ -86,18 +86,21 @@ module.exports.loadUser = async (req, res, authnParams, optionsParams = {}) => {
       maxAge: config.authnCookieMaxAgeMilliseconds,
       httpOnly: true,
       secure: shouldSecureCookie(req),
+      domain: config.cookieDomain,
     });
 
     // After explicitly authenticating, clear the cookie that disables
     // automatic authentication.
     res.clearCookie('pl_disable_auto_authn');
+    res.clearCookie('pl_disable_auto_authn', { domain: config.cookieDomain });
```

Comment:
When clearing cookies, I had to be careful to clear both the unscoped cookie and the one that's scoped to our cookie domain. After a while, this won't be necessary.

Maybe we should have a helper function to encapsulate this and give us a centralized place to document this? Something like `clearCookie(res, name)`.

Reply:
I'm very much in favor of having helper functions so that we can easily check that all cookie handling is being done by the helpers, which will do the right thing. Same for `setCookie()`.

Reply:
I added a `clearCookie` function in ca8ff45. I'm not going to do anything with `setCookie()` just yet to avoid further increasing the scope of this PR.